tw-tasks-core/src/main/java/com/transferwise/tasks/triggering/KafkaTasksExecutionTriggerer.java

```diff
@@ -183,12 +183,10 @@ public void trigger(BaseTask task) {
     // TODO: Future improvement: try to also trigger in the same node, if there is room or more specifically if it is idle (for min latency)
     // TODO: Maybe needs another concurrency control for that. E.g. only trigger in node, when conc < 5, even max conc is 10.
 
-    // We need to give a non-null random key, so triggers will be evenly spread around partitions.
-    // With <null> key, the kafka client will start doing some kind of batch partitioning.
+    final IPartitionKeyStrategy partitionKeyStrategy = taskHandler.getProcessingPolicy(task).getPartitionKeyStrategy();
```

Comment:
If processing policy returns null, we should fall back to the global `IPartitionKeyStrategy` bean right.

Notice, that `taskHandler.getProcessingPolicy(task)` can return null, so we need to avoid null pointer here.

Reply:
Done